src/veins/modules/mobility/traci/subscriptionManagement/PersonSubscriptionManager.h

```diff
@@ -0,0 +1,153 @@
+//
+// Copyright (C) 2006-2017 Christoph Sommer <sommer@ccs-labs.org>
+//
+// Documentation for these modules is at http://veins.car2x.org/
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+
+#ifndef SRC_VEINS_MODULES_MOBILITY_TRACI_SUBSCRIPTIONMANAGEMENT_PERSONSUBSCRIPTIONMANAGER_H_
+#define SRC_VEINS_MODULES_MOBILITY_TRACI_SUBSCRIPTIONMANAGEMENT_PERSONSUBSCRIPTIONMANAGER_H_
+
+#include <set>
+#include <string>
+#include <list>
+
+#include "veins/modules/mobility/traci/subscriptionManagement/SubscriptionManagerBase.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TraCIPerson.h"
+
+namespace veins {
+namespace TraCISubscriptionManagement {
+
+/**
+ * @class PersonSubscriptionManager
+ *
+ * Manages all person subscriptions. You can update it via a subscription
+ * result in the form of a TraCIBuffer or via a manual update but it is
+ * necessary to provide an id_list of active persons then.
+ *
+ * After updating you can request the disappeared and appeared persons via
+ * the class methods.
+ *
+ * IMPORTANT: Call initialize() before using!
+ *
+ * @author Nico Dassler <dassler@hm.edu>
+ */
+class PersonSubscriptionManager: public SubscriptionManagerBase {
+
+public:
+    /**
+     * Constructor.
+     */
+    PersonSubscriptionManager();
+
+    /**
+     * Default destructor.
+     */
+    virtual ~PersonSubscriptionManager() = default;
+
+    /**
+     * Update this PersonSubscriptionManager with the given
+     * active person id list. You can force an update of active
+     * persons using this.
+     *
+     * @param currentlyActivePersonIds A list containing the currently active persons
+     * identified by their ids.
+     *
+     * @return true
+     */
+    bool updateWithList(std::list<std::string>& currentlyActivePersonIds);
+
+    /**
+     * Update this manager with the given buffer containing a subscription
+     * response for a person value.
+     *
+     * This will create new subscriptions for specific persons if the buffer
+     * contained an ID_LIST subscription.
```

Comment:
Better move this into its own named method. Also: I this common behavior of all `TraciSubscriptionManager` specializations? Then it should be part of the common interface!

Reply:
This is in a different method. The comment got removed in my latest changes.

Reply:
Alright.